### DISCUSSION TITLE
How well should I expect "Adams2019" autoscheduler to work for Arm Targets?

### DISCUSSION THREAD

**Discussion by l-oneil** [halide/Halide#5439] [General]
Hi, 

Sorry if the following are dumb questions, I read somewhere, which I can't find now, that the `Adams::2019` auto-scheduler is optimised for x86, so just wondering the following:

As the question states really, my understanding is this auto-scheduler has been optimised for x86? 
Any understanding on it's how good it is on Arm targets?

Also interested in understanding how much work / what work would be required to make it tailored towards Arm/Android devices? 

Kind Regards, 
Liam

**Comment by abadams** [comment]
It should be OK on arm targets, but not great. You'd really have to try it out and see.

To tailor it to arm devices you'd need a training set, which probably means running the random pipeline generator and benchmarking the results on a cluster of arm devices with suitable processors. AWS graviton2 would be the easiest route, but I'm not sure the performance of that would resemble the performance on android phones at all. This is logistically challenging, which is why we haven't done it.

**Reply by ryanstout** [reply]
@abadams not sure if its helpful, but I've got a cluster of 100 armv7 and aarch64 devices (midrange tablet processors) I could run on to train an arm cost model on. (Just need to figure out how and I can commit the samples/weights when done) Thanks!
